static/extensions/SparkyCreations/TimeKeeper.js

```diff
@@ -0,0 +1,211 @@
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: 0
+              },
+            },
+          },
+          {
+            opcode: "deleteAllTimers",
+            blockType: Scratch.BlockType.COMMAND,
+            text: "delete all timers"
+          },
+          {
+            opcode: "deleteTimerNamed",
+            blockType: Scratch.BlockType.COMMAND,
+            text: "delete timer [TIMERNAME]",
+            arguments: {
+              TIMERNAME: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "Timer"
+              },
+            },
+          },
+        ],
+      };
+    }
+  
+    pauseClock() {
+      if (Scratch.vm.runtime.ioDevices.clock && typeof Scratch.vm.runtime.ioDevices.clock.pause === "function") {
+        Scratch.vm.runtime.ioDevices.clock.pause();
+      }
+    }
+  
+    resumeClock() {
+      if (Scratch.vm.runtime.ioDevices.clock && typeof Scratch.vm.runtime.ioDevices.clock.resume === "function") {
+        Scratch.vm.runtime.ioDevices.clock.resume();
+      }
+    }
+  
+    reportTimerInSeconds() {
+      if (Scratch.vm.runtime.ioDevices.clock && typeof Scratch.vm.runtime.ioDevices.clock.projectTimer === "function") {
+        const timer = Scratch.vm.runtime.ioDevices.clock.projectTimer();
+        return timer / 1000;
+      }
+      return 0;
+    }
+  
+    reportTimerInMinutes() {
+      if (Scratch.vm.runtime.ioDevices.clock && typeof Scratch.vm.runtime.ioDevices.clock.projectTimer === "function") {
+        const timer = Scratch.vm.runtime.ioDevices.clock.projectTimer();
+        return timer / (1000 * 60);
+      }
+      return 0;
+    }
+  
+    reportTimerInHours() {
+      if (Scratch.vm.runtime.ioDevices.clock && typeof Scratch.vm.runtime.ioDevices.clock.projectTimer === "function") {
+        const timer = Scratch.vm.runtime.ioDevices.clock.projectTimer();
+        return timer / (1000 * 60 * 60);
+      }
+      return 0;
+    }
+  
+    async reportTimerInInputSeconds(args) {
+      const seconds = args.SECONDS;
+      await new Promise(resolve => setTimeout(resolve, seconds * 1000));
+      if (Scratch.vm.runtime.ioDevices.clock && typeof Scratch.vm.runtime.ioDevices.clock.projectTimer === "function") {
+        return Scratch.vm.runtime.ioDevices.clock.projectTimer();
+      }
+      return 0;
+    }
+  
+    timePaused() {
+      if (Scratch.vm.runtime.ioDevices.clock && Scratch.vm.runtime.ioDevices.clock._projectTimer) {
+        return Scratch.vm.runtime.ioDevices.clock._pausedTime
+      }
+      return 0;
+    }
+  
+    createTimerNamed(args) {
+      const timerName = args.TIMERNAME;
+      const countUpValue = Scratch.Cast.toNumber(args.COUNTUP);
+      if (!this.timers.hasOwnProperty(timerName)) {
+        this.timers[timerName] = 0;
+        this.countUpValues[timerName] = countUpValue;
+        this.intervals[timerName] = setInterval(() => {
+          this.timers[timerName] += this.countUpValues[timerName];
```

Comment:
You shouldn't do this, as many timers all doing this at once will cause lag. 

You should set up a separate function that updates all values at once instead of individually